### ISSUE TITLE
Int32 support needed for maximum tensor scalar version

### ISSUE THREAD

**Issue by Aswinmcw** [tenstorrent/tt-metal#21194]

**Comment by VirdhatchaniKN** [comment]
Update : Issue on Hold, Need inputs

Issue faced : Incorrect results generated with negative scalar input
When the scalar value is negative, the values are stored incorrectly as shown above.
For example, passing the following scalar values returns as mentioned

- -1 --> -2147483647
- -2 --> -2147483646
- -3 --> -2147483645
- -4 --> -2147483644
- -5 --> -2147483643

Would like inputs on this @rdjogoTT , @nvelickovicTT

**Comment by rdjogoTT** [comment]
@VirdhatchaniKN the reason you get -2147483647 when passing in -1 is because -1 is passed in as sign+magnitude, then interpreted as 2's complement. (See bits below)

![Image](https://github.com/user-attachments/assets/72079a11-bb1c-455e-af00-425482cf9abc)

Do you know what the uint bits you are passing to the function as `value` are for -1? Is it `0x80000001` or `0xFFFFFFFF`

**Comment by VirdhatchaniKN** [comment]
Update : 
I tried printing and it gives 0xFFFFFFFF which is 2's complement. I proceeded and have the changes in
Branch : [virdhatchani/max_unary_int32](https://github.com/tenstorrent/tt-metal/pull/23098)
Test file : [tests/ttnn/unit_tests/operations/eltwise/test_unary_maximum.py::test_unary_max_int32_test](https://github.com/tenstorrent/tt-metal/blob/c092dfc13d9bd7db6cc0744bcecbf742d70e3fb4/tests/ttnn/unit_tests/operations/eltwise/test_unary_maximum.py#L14)

**Issue faced : Scalar works only for range [- 2^24,2^24]**
Eg: If we pass -214748360 as scalar (In case of maximum),
- It passes in the composite file as -214748360(as expected) , and then
- In `ttnn/cpp/ttnn/operations/eltwise/unary/unary.cpp`, I have used a template parameter(`{UnaryWithParam{unary_op_type, static_cast<T>(parameter)}}`). 
  - This would pass the scalar value as -214748360, but `UnaryWithParam` accepts the scalar value as float by default according to its struct. Hence, there is loss. 
  - Here, it stores the value as -214748352.0(Which is not expected). 
  - This can be seen by printing the scalar value in `ttnn/cpp/ttnn/operations/eltwise/unary/common/unary_op_utils.cpp` file.

We debugged and the issue is that float has only ~23 bits of precision .So it is rounding the integer value to the closest representable float.

Hence, scalar can be within range [- 2^24,2^24] which is [-16777216, 16777216]. Beyond this, it just rounds up and passes for selected values.

The same issue can be replicated with int supported unary ops using the below code
```
import torch
import ttnn
from tests.ttnn.unit_tests.operations.eltwise.backward.utility_funcs import compare_equal

def test_unary_eq_failing_test(device):
   scalar = -214748360
   torch_input = torch.ones(torch.Size([1, 1, 32, 32]), dtype=torch.int32) * scalar
   golden_function = ttnn.get_golden_function(ttnn.eq)
   golden = golden_function(torch_input, torch.full(torch.Size([1, 1, 32, 32]), scalar), device=device)
   tt_in = ttnn.from_torch(
       torch_input,
       dtype=ttnn.int32,
       device=device,
       layout=ttnn.TILE_LAYOUT,
       memory_config=ttnn.DRAM_MEMORY_CONFIG,
   )
   tt_result = ttnn.eq(tt_in, scalar)
   comp_pass = compare_equal([tt_result], [golden])
   assert comp_pass
```
This would return 0 but expected is 1

**Comment by VirdhatchaniKN** [comment]
[#21194: Add int32 support for Tensor-scalar version of max #23098](https://github.com/tenstorrent/tt-metal/pull/23098)
merged to main
